File: client/go_client/resource_invoke.go
// resource_invoke.go_client
package main

import (
	"context"
	"fmt"
	"github.com/apache/thrift/lib/go/thrift"
	"go_client/gen-go/ResourceService"
)

func ResourceInvoke(host string, port int, method, path, query string) (*ResourceService.ResourceResponse, error) {
	var transport thrift.TTransport
	transport, err := thrift.NewTSocket(fmt.Sprintf("%s:%d", host, port))
	if err != nil {
		return nil, fmt.Errorf("error resolving address: %v", err)
	}
	transport = thrift.NewTBufferedTransport(transport, 8192)

	protocol := thrift.NewTBinaryProtocolTransport(transport)
	client := ResourceService.NewResourceServiceClient(thrift.NewTStandardClient(protocol, protocol))

	if err := transport.Open(); err != nil {
		return nil, fmt.Errorf("error opening transport: %v", err)
	}

	defer transport.Close()

	request := ResourceService.ResourceRequest{
		Method: method,
		Path:   path,
		Query:  query,
	}

	response, err := client.InvokeRequest(context.Background(), &request)
	if err != nil {
		return nil, fmt.Errorf("error invoking request: %v", err)
	}

	return response, nil
}

File: client/py_client/resource_invoke.py
# resource_invoke.py
from thrift.protocol import TBinaryProtocol
from thrift.transport import TSocket, TTransport
from thrift import Thrift
from gen_py.ResourceService import ResourceService
from gen_py.ResourceService.ttypes import ResourceRequest

def ResourceInvoke(host, port, method, path, query):
    transport = TTransport.TBufferedTransport(TSocket.TSocket(host, port))
    protocol = TBinaryProtocol.TBinaryProtocol(transport)
    client = ResourceService.Client(protocol)

    transport.open()

    request = ResourceRequest()
    request.method = method
    request.path = path
    request.query = query

    try:
        response = client.invokeRequest(request)
    except Thrift.TException as tx:
        print("Error invoking request: %s" % tx.message)
        return None

    transport.close()

    return response